solr/core/src/java/org/apache/solr/cluster/scheduler/SolrScheduler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cluster.scheduler;
+
+/**
+ *
+ */
+public interface SolrScheduler {
+
+  void registerSchedulable(Schedulable schedulable);
```

Comment:
How is this method going to be called in practice? Do we assume each node (in the SolrCloud cluster) will register all tasks locally (those that are not `ClusterSingleton`) and that multiple instances are going to run in parallel? Or will a `ClusterSingleton` task be registering other scheduled tasks that as a consequence will only have a single instance running on the cluster?

Reply:
Huh, it looks like I lost some part of this PR during the split ... :) `SolrSchedulerImpl` is a `ClusterSingleton`, at some point I had a section in `CoreContainer` that would instantiate it and add to the list of singletons to be managed. And another section that would scan all registered handlers to see if any one of them was `Schedulable`.

(A side-note: I feel like we keep adding these small hacks just to compensate for the lack of a proper dependency-injection framework... sigh..)